libs/components/modals/src/lib/modules/modal/modal.module.ts

```diff
@@ -36,5 +41,13 @@ import { SkyModalComponent } from './modal.component';
     SkyModalHeaderComponent,
   ],
   entryComponents: [SkyModalHostComponent],
+  providers: [
+    SkyResizeObserverMediaQueryService,
+    SkyResizeObserverService,
+    {
+      provide: SkyMediaQueryService,
+      useExisting: SkyResizeObserverMediaQueryService,
+    },
```

Comment:
Because there are [two injector hierarchies](https://angular.io/guide/hierarchical-dependency-injection#two-injector-hierarchies). With the component providing `SkyMediaQueryService` with `useExisting: SkyResizeObserverMediaQueryService` it will reuse the same instance but redirect to `SkyResizeObserverMediaQueryService` earlier in the injection chain.